engines/suppliers/app/models/suppliers/suppliers_finder.rb

```diff
@@ -0,0 +1,16 @@
+module Suppliers
+  class SuppliersFinder
```

Comment:
In that case let's rename `#find` to `#find_by_group` or maybe just build a repository.